File: 26-remove-duplicates-from-sorted-array/Solution.java
import java.util.HashSet;
import java.util.Set;

public class Solution {
    
    public static void main(String[] args) {
        int[] nums = {0,0,1,1,1,2,2,3,3,4};

        int k = removeDuplicates(nums);

        System.out.println(k);

        for (int i : nums) {
            System.out.print(i);
        }
        

    }

    public static int removeDuplicates(int[] nums) {
        
        int j = 0;
        for(int i  = 0; i < nums.length - 1; i++) {
            if(nums[i] != nums[i + 1]) {
                nums[++j] = nums[i + 1];
            }
        }

        return j + 1;
    }
}

File: 26-remove-duplicates-from-sorted-array/Solution.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:

        j = 0
        for i in range(1, len(nums)):

            if(nums[i] != nums[i - 1]):
                j += 1
            print(i, nums[i])

        return j


nums = [0,0,1,1,1,2,2,3,3,4]
res = Solution.removeDuplicates(Solution, nums)

print("res =", res)

